contracts/tests/MockGatedERC721.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.14;
+import 'hardhat/console.sol';
+
+import '@openzeppelin/contracts/token/ERC721/ERC721.sol';
+import {SismoGated, Attester} from '../core/libs/sismo-gated/SismoGated.sol';
+
+contract MockGatedERC721 is ERC721, SismoGated {
+  uint256 public constant GATED_BADGE_TOKEN_ID = 200001;
+  uint256 public constant GATED_BADGE_MIN_VALUE = 1;
+  mapping(uint256 => bool) private _isNullifierUsed;
+
+  error NFTAlreadyMinted();
+
+  constructor() ERC721('Sismo Gated NFT Contract', 'SGNFT') SismoGated() {}
+
+  function safeMint(
+    address to,
+    uint256 tokenId,
+    bytes[] calldata data
+  )
+    public
+    onlyBadgeOwner(
+      to,
+      GATED_BADGE_TOKEN_ID,
+      GATED_BADGE_MIN_VALUE,
+      HYDRA_S1_ACCOUNTBOUND_ATTESTER,
+      data[0]
+    )
+  {
+    uint256 nullifier = _getNulliferForAddress(to);
+
+    if (isNullifierUsed(nullifier)) {
+      revert NFTAlreadyMinted();
+    }
+
+    _mint(to, tokenId);
+  }
+
+  function safeMintWithTwoGatedBadges(address to, uint256 tokenId, bytes[] calldata data) public {
+    uint256[] memory badgeTokenIds = new uint256[](2);
+    badgeTokenIds[0] = GATED_BADGE_TOKEN_ID;
+    badgeTokenIds[1] = GATED_BADGE_TOKEN_ID + 1;
+
+    uint256[] memory badgeMinimumValues = new uint256[](2);
+    badgeMinimumValues[0] = GATED_BADGE_MIN_VALUE;
+    badgeMinimumValues[1] = GATED_BADGE_MIN_VALUE;
+
+    Attester[] memory attesters = new Attester[](2);
+    attesters[0] = HYDRA_S1_ACCOUNTBOUND_ATTESTER;
+    attesters[1] = HYDRA_S1_ACCOUNTBOUND_ATTESTER;
+
+    checkBadgesOwnership(to, badgeTokenIds, badgeMinimumValues, attesters, data);
+
+    uint256 nullifier = _getNulliferForAddress(to);
+
+    if (isNullifierUsed(nullifier)) {
+      revert NFTAlreadyMinted();
+    }
+
+    _mint(to, tokenId);
+  }
+
+  function safeTransferFrom(
+    address from,
+    address to,
+    uint256 tokenId,
+    bytes memory data
```

Comment:
rename data to sismoProofData 

This is the hard part, they must understand that this is what they will receive via the off-chain prove with sismo flow